tests/queries/0_stateless/03611_null_safe_comparsion.sql

```diff
@@ -0,0 +1,360 @@
+SET enable_analyzer = 1;
+SELECT '=====================================================================';
+SELECT 'Test : ClickHouse NULL-safe comparison';
+SELECT '(1) <=> (IS NOT DISTINCT FROM)';
+SELECT '(2) IS DISTINCT FROM';
+SELECT '=====================================================================';
+SELECT 'Purpose:';
+SELECT '1. Validate behavior of <=> and IS DISTINCT FROM across a wide range of ClickHouse data types and SQL clauses.';
+SELECT '2. Cover numeric, floating, string, enum, date/time, complex types, and NULL / NaN edge cases.';
+SELECT '3. test null-safe comparison in SELECT clause, WHERE, ORDER BY, GROUP BY, HAVING, JOIN, CASE/IF, WINDOW, and subqueries.';
+SELECT '=====================================================================';
+
+SELECT '0. Main table with many types';
+DROP TABLE IF EXISTS 03611_nscmp_tbl;
+CREATE TABLE 03611_nscmp_tbl
+(
+    `c_int8` Int8,
+    `c_int16` Int16,
+    `c_int32` Int32,
+    `c_int64` Int64,
+    `c_uint8` UInt8,
+    `c_uint16` UInt16,
+    `c_uint32` UInt32,
+    `c_uint64` UInt64,
+    `c_float32` Float32,
+    `c_float64` Float64,
+    `c_decimal` Decimal(18, 4),
+    `c_date` Date,
+    `c_datetime` DateTime,
+    `c_dt64` DateTime64(3),
+    `c_string` String,
+    `c_fstring` FixedString(4),
+    `c_enum8` Enum8('a' = 1, 'b' = 2, '' = 0),
+    `c_enum16` Enum16('x' = 100, 'y' = 200, '' = 0),
+    `c_array` Array(Int32),
+    `c_tuple` Tuple(Int32,String),
+    `c_map` Map(String, Int32),
+    `c_nullable` Nullable(Int32),
+    `c_uuid` UUID,
+    `c_ipv4` IPv4,
+    `c_ipv6` IPv6,
+    `c_json` JSON
+)
+ENGINE = MergeTree
+ORDER BY c_int32;
+
+SELECT 'Insert rows containing:';
+SELECT '• normal values';
+SELECT '• NULLs';
+SELECT '• NaN';
+SELECT '• edge numeric boundaries';
+INSERT INTO 03611_nscmp_tbl VALUES
```

Comment:
Could you please order these rows by `c_int32` as you are using that column for `ORDER BY`.